packages/amplify_core/lib/src/types/notifications/push/pinpoint_event_type.dart

```diff
@@ -0,0 +1,15 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+/// {@template amplify_core.push.pinpoint_event_type}
+/// Enum that indicates the different types recorded with Pinpoint
+/// {@endtemplate}
+enum PinpointEventType {
+  notificationOpened('opened_notification'),
+  backgroundMessageReceived('received_background'),
+  foregroundMessageReceived('received_foreground');
+
+  const PinpointEventType(this.name);
+
+  final String name;
```

Comment:
Don't think `name` needs to be overridden on these, unless it's important for native code somehow

Reply:
`name` is consumed when recording an event and needs to of a specific format hence overriding it. 
https://github.com/aws-amplify/amplify-flutter/blob/hui/feat/push-notifications-rebase/packages/notifications/push/amplify_push_notifications_pinpoint/lib/src/pinpoint_provider.dart#L126